### PR TITLE
[Product Creation AI v2] Package photo view and Short description field

### LINKED ISSUE #13246: View to display the selected product image.
image

- Pass selected package photo to preview screen
- Display package photo in preview screen.

### LINKED ISSUE #13254: Add text field for short description
- Add a text field for short description
- Move package photo view below name and description section

Internal - p1720140316203299/1720090796.407329-slack-C03L1NF1EA3

| Full screen | Short description field area  |
|--------|--------|
| image | image |

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #13254
Closes: #13246

## Description

Adds package photo view and short description field to the product preview screen.

Changes
- 

## Steps to reproduce

1. Create a JN site with Jetpack AI plugin
2. Turn on `productCreationAIv2M1` feature flag
3. Build and run the app
4. Navigate to the products tab
5. Tap `+` button on top right
6. Tap "Create a product with AI"
7.  Select a package photo
8. Tap "Generate Product Details" and you will land in the new product preview screen
9. Validate that the selected product is displayed in the preview screen. Try viewing and removing the selected photo and validate that it works as expected. 
10. Validate that the short description field is added.

## Screenshots

| Light | Dark |
|--------|--------|
| Simulator Screenshot - iPhone 15 Pro Max - 2024-07-05 at 12 14 49 | Simulator Screenshot - iPhone 15 Pro Max - 2024-07-05 at 12 14 59 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 1]
```diff
@@ -259,6 +298,89 @@ private extension ProductDetailPreviewView {
     }
 }
 
+// MARK: - Package photo view
+
+private extension ProductDetailPreviewView {
+    struct PackagePhotoView: View {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: ❓ I see that this view is quite identical to the one on the starting page. Should we reuse it and add an option to show/hide the subtitle and the Replace photo option instead?

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 2]
```diff
@@ -316,14 +438,19 @@ fileprivate extension ProductDetailPreviewView {
         )
         static let productNameAndDescription = NSLocalizedString(
             "productDetailPreviewView.productNameAndDescription",
-            value: "Name & Description",
-            comment: "Title of the name and description fields on the add product with AI Preview screen."
+            value: "Name, Summary & Description",
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Should we change the key to this string as well? I still see the old text "Name & Description", I suppose this was due to the old translation being used.

**Review by itsmeichigo — APPROVED** [review]
This works as described! 🚢

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 2 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/13255/commits/e1ef76bc84f3ac53e057391b15c55d04f53996c9

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift]: Good suggestion. Done in https://github.com/woocommerce/woocommerce-ios/pull/13255/commits/2c4bf7a80644dbfac0b742d2069ac05fd303084c and https://github.com/woocommerce/woocommerce-ios/pull/13255/commits/7524f0f5f962872bfcd1749b754cedba7bae8f1d.

**Pull request merged**